jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/S3VFS.kt

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package software.aws.toolkits.jetbrains.services.s3
+
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileListener
+import com.intellij.openapi.vfs.VirtualFileSystem
+import software.amazon.awssdk.services.s3.S3Client
+
+class S3VFS(
+    val client: S3Client
+) : VirtualFileSystem() {
+    override fun getProtocol(): String = "S3"
+
+    override fun isReadOnly(): Boolean = false
+
+    override fun refresh(asynchronous: Boolean) {}
+
+    override fun addVirtualFileListener(listener: VirtualFileListener) {}
+
+    override fun findFileByPath(path: String): VirtualFile? = null
+
+    override fun refreshAndFindFileByPath(path: String): VirtualFile? = null
+
+    override fun removeVirtualFileListener(listener: VirtualFileListener) {}
+
+    @Throws(UnsupportedOperationException::class)
```

Comment:
I think it's safe to leave out the `@Throws` annotations. They're only used for generating Java function signatures when you're [calling Kotlin code from Java](https://kotlinlang.org/docs/reference/java-to-kotlin-interop.html#checked-exceptions), and I don't see any of that here.